pkg/cluster/api/pdapi.go

```diff
@@ -530,23 +545,19 @@ func (pc *PDClient) DelPD(name string, retryOpt *utils.RetryOption) error {
 
 func (pc *PDClient) isSameState(host string, state metapb.StoreState) (bool, error) {
 	// get info of current stores
-	stores, err := pc.GetStores()
+	storeInfo, err := pc.GetCurrentStore(host)
 	if err != nil {
 		return false, err
 	}
+	if storeInfo == nil {
```

Comment:
If storeInfo == nil, the err will not be nil. So I think this code is not necessary。